src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -69,59 +59,53 @@ export class WorkspaceSavedObjectsClientWrapper {
 
     return [permission];
   }
-  private async validateSingleWorkspacePermissions(
-    workspaceId: string | undefined,
-    request: OpenSearchDashboardsRequest,
-    permissionMode: WorkspacePermissionMode | WorkspacePermissionMode[]
-  ) {
-    if (!workspaceId) {
-      return;
-    }
-    const validateResult = await this.permissionControl.validate(
-      request,
-      {
-        type: WORKSPACE_TYPE,
-        id: workspaceId,
-      },
-      this.formatWorkspacePermissionModeToStringArray(permissionMode)
-    );
-    if (!validateResult?.result) {
-      throw generateWorkspacePermissionError();
-    }
-  }
 
-  private async validateMultiWorkspacesPermissions(
-    workspaces: string[] | undefined,
+  private async validateObjectsPermissions(
+    objects: Array<Pick<SavedObject, 'id' | 'type'>>,
     request: OpenSearchDashboardsRequest,
     permissionMode: WorkspacePermissionMode | WorkspacePermissionMode[]
   ) {
-    if (!workspaces) {
-      return;
-    }
-    for (const workspaceId of workspaces) {
+    // PermissionMode here is an array which is merged by workspace type required permission and other saved object required permission.
+    // So we only need to do one permission check no matter its type.
+    for (const { id, type } of objects) {
       const validateResult = await this.permissionControl.validate(
         request,
         {
-          type: WORKSPACE_TYPE,
-          id: workspaceId,
+          type,
+          id,
         },
         this.formatWorkspacePermissionModeToStringArray(permissionMode)
       );
       if (!validateResult?.result) {
-        throw generateWorkspacePermissionError();
+        return false;
       }
+      return true;
+    }
+  }
+
+  // validate the permission of multi workspace type saved object
```

Comment:
```suggestion
  // validate if the `request` has the specified permission(`permissionMode`) to the given `workspaceIds`
```
Is this what the function do?